include/unifex/tracing/async_stack-inl.hpp

```diff
@@ -67,6 +67,16 @@ popAsyncStackFrameCallee(unifex::AsyncStackFrame& calleeFrame) noexcept {
   calleeFrame.stackRoot = nullptr;
 }
 
+inline void popAsyncStackFrameFromCaller(
+    [[maybe_unused]] unifex::AsyncStackFrame& callerFrame) noexcept {
+  auto root = tryGetCurrentAsyncStackRoot();
+  assert(root != nullptr);
```

Comment:
why call it `try` if it cannot be `nullptr`?

Reply:
The current async stack root is allowed to be `nullptr` in general, but the stack is in a bad state if it's null in this case. The alternative way to read the current stack root is `getCurrentAsyncStackRoot()`, but that returns a reference; if the stack is corrupt at this point, we can't meaningfully assert that a reference is valid, but we _can_ assert that a nullable pointer isn't null.